src/FingerprintGenerator/UserAgent.php

```diff
@@ -3,12 +3,27 @@
 
 class UserAgent implements FingerprintGeneratorInterface
 {
+    /** @var string */
+    private $userAgent;
+
+    /**
+     * @param string $userAgent
+     */
+    public function __construct($userAgent = null)
+    {
+        if($userAgent !== null) {
+            $srv = array_merge(array('HTTP_USER_AGENT' => ''), $_SERVER);
```

Comment:
Allowing to pass $userAgent is great, but merging whole $_SERVER does not make much of sense to me.

What about...

```
if ($userAgent !== null) {
    $userAgent = (isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : '');
}